src/Automation/SnapshotCommand.cs

```diff
@@ -14,30 +14,33 @@
 namespace Axe.Windows.Automation
 {
     /// <summary>
-    /// Class to take a snapshot (via ShapshotCommand.Execute). Can only be successfully called after
-    /// a successful call to StartCommand.Execute
+    /// Class to take a snapshot (via SnapshotCommand.Execute). 
     /// </summary>
     static class SnapshotCommand
     {
         /// <summary>
         /// Execute the Start command. Used by both .NET and by PowerShell entry points
         /// </summary>
         /// <param name="config">A set of configuration options</param>
-        /// <param name="outputFileHelper"/>
-        /// <param name="resultsAssembler"/>
+        /// <param name="scanTools">A set of tools for writing output files,
+        /// creating the expected results format, and finding the target element to scan</param>
         /// <returns>A SnapshotCommandResult that describes the result of the command</returns>
-        public static ScanResults Execute(Config config, IOutputFileHelper outputFileHelper, IScanResultsAssembler resultsAssembler)
+        public static ScanResults Execute(Config config, IScanTools scanTools)
         {
             return ExecutionWrapper.ExecuteCommand<ScanResults>(() =>
             {
-                if (outputFileHelper == null) throw new ArgumentNullException(nameof(outputFileHelper));
-                if (resultsAssembler == null) throw new ArgumentNullException(nameof(resultsAssembler));
+                if (scanTools == null) throw new ArgumentNullException(nameof(scanTools));
+                if (scanTools.OutputFileHelper == null) throw new ArgumentNullException(nameof(scanTools.OutputFileHelper));
```

Comment:
>OutputFileHelper [](start = 30, length = 16)

These checks already exist in the ScanTools class. If we posit that IScanTools *must* provide a non-null value here, then we can       omit the extra checks.

Reply:
This class only deals with the IScanTools abstraction and therefore cannot make that assumption.

Reply:
My point is that we can make it a requirement within the interface that the implementation has valid values. There's a proposal to provide non-nullable reference types in the future, maybe we ought to use it when it's available

---
In reply to: [294516127](https://github.com/microsoft/axe-windows/pull/57#discussion_r294516127) [](ancestors = 294516127)